packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -138,4 +104,102 @@ export class TokenAuthorizer extends Authorizer implements IAuthorizer {
 
     this.restApiId = restApi.restApiId;
   }
-}
\ No newline at end of file
+
+  protected abstract getResource<T extends LambdaAuthorizerProps>(props: T): CfnAuthorizer;
```

Comment:
cc @eladb @RomainMuller - do you guys know if this is ok?

Reply:
Yes - `get*` would collide with a Java getter for a property named `*`, so we prefer to avoid that whenever possible. `jsii` may even hard-block that altogether.

Reply:
Further, jsii will fail the build once this type is exported.

Reply:
This won't be exported, so there's no issue. I'll rename the method to avoid issues with `Java`.

Reply:
Re: generics.

AFAIK, we've managed to come this far without generics anywhere on the CDK code. I'd rather keep it that way, unless there's a compelling reason. And I don't see this PR creating that compelling case. 

I've taken a stab at re-arranging the code to avoid using generics. see attached git patch file. (I couldn't push a new branch into your fork, so this was the next best way to share changes).

Reply:
Ugh - file attachments to comments are not working for some reason. Here's the gist instead - https://gist.github.com/nija-at/80d71b39f5212b67ba496055eb4e6af5